framework/topology.cpp

```diff
@@ -27,6 +27,72 @@
 #   include <windows.h>
 #endif
 
+namespace {
+struct TopologyDetector
+{
+    TopologyDetector();
+    void detect(const LogicalProcessorSet &enabled_cpus);
+    void sort();
+
+private:
+    bool setup_cpuid_detection();
+    bool detect_via_cpuid(Topology::Thread *info);
+    bool detect_via_os(Topology::Thread *info);
```

Comment:
I find confusing the names `detect_via_os()` and `detect_topology_via_os()`, like if they do the same thing. Either way, this is only a comment and no changes are requested.

Reply:
Not necessarily. "detect_via_os" is "detect everything", not just the topology (see other functions).